src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -27,10 +31,10 @@
                                            #"^([a-zA-Z0-9\.\-]+)/([a-zA-Z0-9\.\-]+)$"
                                            line)]
     {:namespace namespace :service-name service-name}
-    (throw (IllegalArgumentException.
-            (str "Invalid line in bootstrap config file:\n\n\t"
-                 line
-                 "\n\nAll lines must be of the form: '<namespace>/<service-fn-name>'.")))))
+    (throw+ {:type :illegal-argument
```

Comment:
this is a nitpick, but I might use a type that was more specific here; maybe something like `::bootstrap-parse-error`.  The only reason we used a generic type (`IllegalArgumentException`) above is because we didn't want to have to create a new Java Exception class.